src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,123 @@
+        } catch (SQLException e) {
+            throw new DataProcessingException(CREATE_EXCEPTION_MESSAGE, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_BY_ID_SQL_QUERY)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(createBookFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(FIND_BY_ID_EXCEPTION_MESSAGE, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> allBooks = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(FIND_ALL_SQL_QUERY)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                allBooks.add(createBookFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(FIND_ALL_EXCEPTION_MESSAGE, e);
+        }
+        return allBooks;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(UPDATE_SQL_QUERY)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException(NO_CHANGES_EXCEPTION_MESSAGE);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(UPDATE_EXCEPTION_MESSAGE, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(DELETE_SQL_QUERY)) {
+            statement.setLong(1, id);
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException(NO_CHANGES_EXCEPTION_MESSAGE);
+            }
+            return affectedRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException(DELETE_EXCEPTION_MESSAGE, e);
+        }
+    }
+
+    private Book createBookFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getLong(1);
+        String title = resultSet.getString(2);
+        BigDecimal price = BigDecimal.valueOf(resultSet.getDouble(3));

```

Comment:
You can fetch BigDecimal like this:
```suggestion
        BigDecimal price = resultSet.getObject("price", BigDecimal.class);
```